src/hotspot/share/oops/fieldStreams.hpp

```diff
@@ -193,9 +193,74 @@ class AllFieldStream : public FieldStreamBase {
   AllFieldStream(const InstanceKlass* k):      FieldStreamBase(k->fieldinfo_stream(), k->constants()) {}
 };
 
+template<typename FieldStreamType>
```

Comment:
It would be good to say what this class does and what the template parameter is for.